core/src/main/scala/akka/persistence/r2dbc/R2dbcSettings.scala

```diff
@@ -244,26 +247,64 @@ final class R2dbcSettings private (
    * The journal table and schema name with data partition suffix for the given slice. When number-of-partitions is 1
    * the table name is without suffix.
    */
-  def journalTableWithSchema(slice: Int): String = {
-    if (numberOfDataPartitions == 1)
-      journalTableWithSchema
-    else
-      s"${journalTableWithSchema}_${dataPartition(slice)}"
-  }
+  def journalTableWithSchema(slice: Int): String =
+    resolveTableName(journalTableWithSchema, slice)
 
+  /**
+   * The snapshot table and schema name without data partition suffix.
+   */
   val snapshotsTableWithSchema: String = schema.map(_ + ".").getOrElse("") + snapshotsTable
+
+  /**
+   * The snapshot table and schema name with data partition suffix for the given slice. When number-of-partitions is 1
+   * the table name is without suffix.
+   */
+  def snapshotTableWithSchema(slice: Int): String =
+    resolveTableName(snapshotsTableWithSchema, slice)
```

Comment:
What do you mean? This is the new method, which should always be used. The old `snapshotsTableWithSchema` without slice parameter could become private, but I didn't want to break bin compat (`@InternalStableApi`), but maybe worth doing that?

Reply:
Sorry, maybe I was unclear, what I meant is that the consuming code calls this, which generates a new string for every call instead of using the `allSnapshotTablesWithSchema` lookup table defined below. Probably not a big thing but seems wasteful when we anyway have the lookup table.

Reply:
Ah, I see now the lookup table is the reverse, name to slice

Reply:
I see what you mean, but `allSnapshotTablesWithSchema` isn't a lookup for each of the 1024 slices. It contains one entry per data partition, where the slice is the lower slice for that data partition.
E.g. `snapshotTableWithSchema(slice = 17)` can't be found easily in `allSnapshotTablesWithSchema`.

To make it a full lookup table it would have to contain 1024 entries, with the slice as key. Might be possible with some other structure such as binary search but not sure that is worth it. Then I'm more worried about all string construction of the full sql statements that now happen on each use. We could introduce a cache for them, and that would also solve this because the table names are always used within such sql.

`allSnapshotTablesWithSchema` was mostly added because in tests we delete all tables before running.